experiments/Marmot/Venn/tikzlibraryvenn.code.tex

```diff
@@ -0,0 +1,174 @@
+%
+% See the file doc/generic/pgf/licenses/LICENSE for more details.
+% Big thanks go to JouleV and samcarter for feedback and suggestions!
+\ProvidesFileRCS{tikzlibraryvenn.code.tex}
+\usetikzlibrary{backgrounds,calc}
+\newif\ifvenn@reversed
+\newif\ifvenn@labelmath
+\venn@reversedfalse
+\tikzset{reuse path/.code={%
+    \pgfsyssoftpath@setcurrentpath{#1}}
+}% unfortunately this is *not* the definition of the `use path` key that has
+% been added to the pgfmanual!
+\tikzset{current reverse clip/.style={insert path={{% https://tex.stackexchange.com/a/127045
+   (current bounding box.south east)  -|
+  (current bounding box.north west) -| cycle
+  }}},
+  current reverse clip'/.style={insert path={{% https://tex.stackexchange.com/a/127045
+   (current bounding box.south east)  -|
+  (current bounding box.north west) -| cycle
+  }}},
+  even odd clip/.code={% https://tex.stackexchange.com/a/76216
+  \pgfseteorule}}
+\tikzset{Venn diagram/.style={/utils/exec=\tikzset{venn/.cd,#1},
+ execute at begin scope={
+ \pgfmathtruncatemacro{\pgfutil@tmpD}{dim(\pgfkeysvalueof{/tikz/venn/labels})}
+ \ifnum\pgfutil@tmpD<\pgfkeysvalueof{/tikz/venn/number of sets}
+ \tikzerror{Less labels then Venn sets.}%
+ \fi
+ \pgfmathsetmacro{\pgfutil@tmpA}{{\pgfkeysvalueof{/tikz/venn/labels}}[0]}
+ \pgfmathsetmacro{\pgfutil@tmpB}{{\pgfkeysvalueof{/tikz/venn/labels}}[1]}
+ \pgfmathsetmacro{\pgfutil@tmpC}{{\pgfkeysvalueof{/tikz/venn/labels}}[2]}
+ \begin{scope}[local bounding box=Venn Box]
+ \foreach \Set [count=\Num starting from 0] in {1,...,\pgfkeysvalueof{/tikz/venn/number of sets}}
+ {\pgfmathsetmacro{\mylabel}{{\pgfkeysvalueof{/tikz/venn/labels}}[\Num]}
+ \ifvenn@labelmath
+  \draw[/tikz/venn/lines,/tikz/venn/circle \mylabel/.try] ({\pgfkeysvalueof{/tikz/venn/offset angle}+240-30*\pgfkeysvalueof{/tikz/venn/number of sets}+
+  360*\Num/\pgfkeysvalueof{/tikz/venn/number of
+  sets}}:\pgfkeysvalueof{/tikz/venn/distance})  
+  coordinate[label={[anchor=center,/tikz/venn/all labels,/tikz/venn/label \mylabel/.try]:$\mylabel$}]
+  (\pgfkeysvalueof{/tikz/venn/center prefix}\mylabel) circle[radius=\pgfkeysvalueof{/tikz/venn/radius}];
+ \else 
+  \draw[/tikz/venn/lines,/tikz/venn/circle \mylabel/.try] ({\pgfkeysvalueof{/tikz/venn/offset angle}+240-30*\pgfkeysvalueof{/tikz/venn/number of sets}+
+  360*\Num/\pgfkeysvalueof{/tikz/venn/number of
+  sets}}:\pgfkeysvalueof{/tikz/venn/distance}) 
+  coordinate[label={[anchor=center,/tikz/venn/all labels,/tikz/venn/label \mylabel/.try]:\mylabel}]
+  (\pgfkeysvalueof{/tikz/venn/center prefix}\mylabel) circle[radius=\pgfkeysvalueof{/tikz/venn/radius}];
+ \fi}
+ \tikzset{venn/custom foreground paths}
+ \end{scope}
+ \pgf@venn@codes
+ \draw[/tikz/venn/frame] ([xshift=-\pgfkeysvalueof{/tikz/venn/frame margin},
+ yshift=-\pgfkeysvalueof{/tikz/venn/frame margin}]Venn Box.south west) 
+ rectangle ([xshift=\pgfkeysvalueof{/tikz/venn/frame margin},
+ yshift=\pgfkeysvalueof{/tikz/venn/frame margin}]Venn Box.north east); 
+},on background layer},
```

Comment:
Is this where the enormous `execute at begin scope` ends?  I can't really see for the lack of indentation.  Also when doing this you have to somehow make sure that `Venn diagram` can only be used as a scope or picture option.